### ISSUE TITLE
Dynamic Codepipeline actions

### ISSUE THREAD

**Issue by lengland** [terraform-providers/terraform-provider-aws#6211]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I think it would be useful if we could make a dynamic list of actions inside a single stage using Terraform using `count.index` and a data map. This would allow defining a list of source repositories or list of build steps and have them run in parallel. 

### New or Affected Resource(s)

* aws_codepipeline

### Potential Terraform Configuration

```hcl
  stage {
    name = "GitHub"
    action {
      count = "${length(some_var)}"
      name = "soucecode-${count.index}"
      category = "Source"
      owner = "ThirdParty"
      provider = "GitHub"
      version = "1"
      output_artifacts = ["soucecode-${count.index}"]

      configuration {
        Owner = "${var.github_org}"
        Repo = "${var.repo_list[count.index].repo}" #Pseudo-code
        Branch = "${var.repo_list[count.index].branch}" #Pseudo-code
        OAuthToken = "${var.github_token}"
      }
    }
  }
```

### References

None

**Comment by bflad** [comment]
Hi @lengland 👋 

Handling dynamic sub-resource configurations should be covered in the upcoming Terraform 0.12 release using a new `for` and `for_each` syntax within the configuration language (HCL). Further tracking for that specific feature request can be found here: https://github.com/hashicorp/terraform/issues/7034

A high level preview of this feature and others coming in that version can be found in: https://www.hashicorp.com/blog/terraform-0-1-2-preview

Can you take a look at both of those and see if they cover what you are looking for here? Thanks!

**Comment by lengland** [comment]
@bflad Thanks for the update, that seems like its on the right track. I'll follow the feature request because that sounds exactly like what I am requesting.
I guess I'll find out for sure when I get my hands on TF 0.12 :)

**Comment by bflad** [comment]
Sounds good! I'll close this out. The 0.12 branch was merged into master this week so getting closer!

**Comment by jcarlson** [comment]
Is there a recommended work around to achieve this in Terraform 0.11.x? Since 0.12 isn't final yet, and I'm not sure how soon after it is my team will be able to upgrade. I probably need to find a suitable workaround in the short term.
